src/azure-cli/setup.py

```diff
@@ -53,7 +53,7 @@
     'antlr4-python3-runtime~=4.7.2',
     'azure-appconfiguration~=1.1.1',
     'azure-batch~=10.0.0',
-    'azure-cli-core=={}.*'.format(VERSION),
+    'azure-cli-core=={}.*'.format(".".join(VERSION.split(".")[:3])),
```

Comment:
`VERSION` can be in the format of `2.16.0` or `2.16.0.devN`. Both cases will result in `azure-cli-core==2.16.0.*`.

Reply:
It seems `2.16.0.*` can match with both `2.16.0` and `2.16.0.devN`, but `2.16.0.devN.*` cannot match `2.16.0.devN` in pip 20.3.